### ISSUE TITLE
change `git prune-branches` to delete branches without changes

### ISSUE THREAD

**Issue by kevgo** [git-town/git-town#2021]
### situation

- I create a lot of branches
- the branches take a long time to get reviewed
- reviews get shipped by other people
- these people sometimes don't delete the remote branch

### problem

- lots of old branches containing no changes accumulate

### solution

The command "git prune-branches" goes through each branch on the local repo and:

- syncs it
- determines if it is empty: `git diff <main branch>` is empty
- if empty, deletes the branch

### action items

- [ ] normal `prune-branches` command prunes local branches only
- [ ] `prune-branches --remote` also prunes all branches that exist at the remote?

**Comment by koppor** [comment]
Would that also help with deletion of squash-merged branches? Currently, deleting squash-merged branches only works it the remote branch is removed. Otherwise, these branches also pile-up (aren't they?).

**Comment by kevgo** [comment]
Yes. The new behavior deletes all branches for which `git diff <parent-branch>` is empty. That's the case for squash-merged branches.

**Comment by koppor** [comment]
I was asking, because in the case of squash-merge a) the parent branch is gone and b) there are more commits to `origin/main` after the squash merge, so the `git diff main` is not empty.

**Comment by kevgo** [comment]
Thanks for clarifying. The behavior I have in mind is that if the updated branch has no diff compared to the main branch, it would be deleted. From the "solution" section above:

> The command "git prune-branches" goes through each branch on the local repo and:
>
> - syncs it
> - determines if it is empty: git diff <main branch> is empty
> - if empty, deletes the branch

If the parent is unknown, then it would assume the main branch as the parent. Would this work for your use case or do you foresee any other complications? Thanks for the feedback!

**Comment by koppor** [comment]
TL;DR: My comments are due to misunderstandings. Can be deleted (or converted to documentation).

Think, I still did not get the full concepts of `git-town`. It "enforces" somehow a "clean" git usage. My old habit was to pile up local branches without any PR. Thus, without any documentation... I need to change that.

In a proper git workflow usage, `git-town prune branches` will do the job in my case.

Your issue is a nice-addition for your use case.

Let me share screenshots, why `git-town prune branches` does the job well. (Assumption: "proper" usage of remote repositories).

Example: https://github.com/JabRef/jabref/pull/9726

State of the repository:

![image](https://user-images.githubusercontent.com/1366654/230319904-9594aa3f-2e63-4ac9-8641-ed695248487a.png)
[...]
![image](https://user-images.githubusercontent.com/1366654/230319802-eeb586ae-b4f2-4df2-93fd-1011e4387edd.png)

git-town does its job niceley:

![image](https://user-images.githubusercontent.com/1366654/230321229-52872f4f-e5a3-4402-b6fc-14ed6076456f.png)


Feel free to remove all these comments to keep the issue tracker clean.

I can also try to submit a PR to enhance the documentation.

**Comment by kevgo** [comment]
> I can also try to submit a PR to enhance the documentation.

That would be great, thanks! :slightly_smiling_face:
